pxr/usd/usdMtlx/parser.cpp

```diff
@@ -239,6 +327,45 @@ ShaderBuilder::AddProperty(
         metadata[SdrPropertyMetadata->ImplementationName] = j->second;
     }
 
+    if (!isOutput) {
+        ParseMetadata(metadata, SdrPropertyMetadata->Label, element, "uiname");
+        ParseMetadata(metadata, SdrPropertyMetadata->Help, element, "doc");
+        ParseMetadata(metadata, SdrPropertyMetadata->Page, element, "uifolder");
+
+        ParseMetadata(metadata, _tokens->uimin, element);
+        ParseMetadata(metadata, _tokens->uimax, element);
+        ParseMetadata(metadata, _tokens->uisoftmin, element);
+        ParseMetadata(metadata, _tokens->uisoftmax, element);
+        ParseMetadata(metadata, _tokens->uistep, element);
+        ParseMetadata(metadata, _tokens->unit, element);
+        ParseMetadata(metadata, _tokens->unittype, element);
+        ParseMetadata(metadata, _tokens->defaultgeomprop, element);
+
+        if (!metadata.count(SdrPropertyMetadata->Help) &&
+             metadata.count(_tokens->unit)) {
+            // The unit can be helpful if there is no documentation.
```

Comment:
Not sure what `unittype` is, but would it also be useful?  Seems like `defaultgeomprop` would also be handy to provide?

Reply:
I see only two places in MaterialX where unittype is defined. There are a few angles defined as being "unittype = radian" and one thin_film thickness being tagged with "unittype = nanometer". I suspect ST coordinates could be in meters in some future, but we are not there yet.

Reply:
Oh hmmm...   these will want to be **very prominently** documented, as they diverge from USD norms.  All angles are degrees in USD schemas, and linear units are generally relative to stage `metersPerUnit` .  So yeah, I'd double-down on wanting to fold unittype into the description, if the nodes that use them don't already provide documentation that mentions them!